src/Npgsql/Internal/NpgsqlConnector.Auth.cs

```diff
@@ -71,9 +71,11 @@ async Task AuthenticateCleartext(string username, bool async, CancellationToken
     async Task AuthenticateSASL(List<string> mechanisms, string username, bool async, CancellationToken cancellationToken)
     {
         // At the time of writing PostgreSQL only supports SCRAM-SHA-256 and SCRAM-SHA-256-PLUS
-        var supportsSha256 = mechanisms.Contains("SCRAM-SHA-256");
-        var supportsSha256Plus = mechanisms.Contains("SCRAM-SHA-256-PLUS");
-        if (!supportsSha256 && !supportsSha256Plus)
+        var serverSupportsSha256 = mechanisms.Contains("SCRAM-SHA-256");
+        var clientSupportsSha256 = serverSupportsSha256 && Settings.ChannelBinding != ChannelBinding.Require;
+        var serverSupportsSha256Plus = mechanisms.Contains("SCRAM-SHA-256-PLUS");
+        var clientSupportsSha256Plus = serverSupportsSha256Plus && Settings.ChannelBinding != ChannelBinding.Disable;
+        if (!clientSupportsSha256 && !clientSupportsSha256Plus)
```

Comment:
We may want to be a bit more informative with the exception message (and possibly have multiple), i.e. "couldn't connect because ChannelBinding is set to Require but the server doesn't support SCRAM-SHA-256-PLUS (and similar for Disable).